spec/teacher_spec.rb

```diff
@@ -0,0 +1,17 @@
+require 'rspec'
+require_relative '../teacher'
+
+describe Teacher do
+  context 'testing the constructor' do
+    it 'should return a Teacher object' do
+      teacher = Teacher.new(34, 'Jane Doe', 'Math')
+      expect(teacher).to be_an_instance_of(Teacher)
+    end
+  end
+  context 'testing the can_use_services? method' do
+    it 'should return true' do
+      teacher = Teacher.new(34, 'Jane Doe', 'Math')
+      expect(teacher.can_use_services?).to eq(true)
+    end
+  end
```

Comment:
- [ ] Great job for testing most of the classes but I think you forgot to test the classes that you created while implementing the Decorator design pattern so please add tests for nameable, decorator, capitalize, and Trimmer classes since they are mentioned in the requirements so kindly test them 🤩🔥♻🚀✨💯  


![image](https://github.com/lRebornsl/school-library/assets/58553711/e106dd6e-2347-4cd8-b603-f8b60f3737fe)